### PR TITLE
Fixes #7707, #7708 health snapshot, custom report

### LINKED ISSUE #7707: feat: Convert portal_patient_report.php file to  use twig template for module writers to extend
As a developer I would like to customize the html contents of the portal_patient_report.php file.  In order to do this the file should be converted to using twig as well as having an event that lets me filter the twig data properties in order to customize the report settings.

### LINKED ISSUE #7708: feat: pull patient portal health snapshot into modular twig/filter event for customization
As a developer I would like to customize the portal health snapshot interface with custom data as well as override parts of the health snapshot interface using twig.

### PR DESCRIPTION
**Pull request by adunsulag**
Made it so the custom report foudn in patient_report.php uses twig and also has a filter event for customizing the data that gets passed into the report selection system.  This allows module writers to extend the interface and add data or hide sections as needed.

Did the same thing with the health snapshot to allow pieces of the snapshot to be overridden by twig templates.

This work was done to facilitate a module that wanted to have flags to hide/show different parts of the reports and health snapshot as they pilot test these areas with their patients.

Fixes #7707
Fixes #7708

### CODE REVIEW

File: portal/home.php [hunk 1]
```diff
@@ -31,6 +31,8 @@
 use Twig\Error\LoaderError;
 use Twig\Error\RuntimeError;
 use Twig\Error\SyntaxError;
+use OpenEMR\Events\PatientReport\PatientReportFilterEvent;
+use OpenEMR\Common\Logging\SystemLogger;
```
Comment (bradymiller) [inline portal/home.php:35]: alphabetize ordering

File: portal/report/portal_patient_report.php [hunk 2]
```diff
@@ -39,6 +41,11 @@
 require_once("$srcdir/patient.inc.php");
 
 use OpenEMR\Core\Header;
+use OpenEMR\Common\Logging\SystemLogger;
+use OpenEMR\Controllers\Portal\PortalPatientReportController;
+use OpenEMR\Common\Twig\TwigContainer;
+use OpenEMR\Events\PatientReport\PatientReportFilterEvent;
+use Twig\Error\SyntaxError;
```
Comment (bradymiller) [inline portal/report/portal_patient_report.php:48]: alphabetize

File: src/Controllers/Portal/PortalPatientReportController.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+use OpenEMR\Common\Database\QueryUtils;
+
+class PortalPatientReportController
+{
+    public function getDocuments($pid)
+    {
+
+        // show available documents
+        $db = $GLOBALS['adodb']['db'];
```
Comment (bradymiller) [inline src/Controllers/Portal/PortalPatientReportController.php:25]: is above $db variable being used?

File: templates/portal/partial/reports/patient_report/_issues_encounters.html.twig [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        {% for disptype,issuesCategory in issues %}
+        <tr>
+            <td colspan='4' class='bold'><b>{{ issuesCategory.display|text }}</b></td>
+        </tr>
+        {% for issue in issuesCategory.issues %}
+        <tr class='text'>
+            <td>&nbsp;</td>
+            <td>
+                <label class="d-block" for='issue_{{ issue.id|attr }}'>
+                    <input type='checkbox' name='issue_{{ issue.id|attr }}' id='issue_{{ issue.id|attr }}' class='issuecheckbox' value='/{{ issue.encounter|join("/")|text }}' />
```
Comment (bradymiller) [inline templates/portal/partial/reports/patient_report/_issues_encounters.html.twig:25]: attr for the value attribute

File: templates/portal/partial/reports/patient_report/_issues_encounters_form.html.twig [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+                               id='{{ encounter.formdir|attr }}_{{ encounter.form_id|attr }}'
+                               value='{{ encounter.encounter|attr }}' class='encounter'>{{ encounter.display|text }} ({{ form.date|date("Y-m-d") }})
+                    </label>
+                    <div class='encounter_forms'>
+                        {% for formName,formList in encounter.forms %}
+                        {% for form in formList %}
+                        <label>
+                            <input type="checkbox" name="{{ form.formdir|attr }}_{{ form.form_id|attr }}"
+                                   id="{{ form.formdir|attr }}_{{ form.form_id|attr }}"
+                                   value="{{ form.encounter|attr }}" class="encounter_form">{{ form.display|attr }} ({{ form.date|date("Y-m-d")|attr }})
```
Comment (bradymiller) [inline templates/portal/partial/reports/patient_report/_issues_encounters_form.html.twig:36]: only need text for the form.display and form.date

**Comment by bradymiller** [comment]
@adunsulag , just noted several minor issues. Otherwise code looks good for bringing in.

File: portal/home.php (hunk 1 above)
Comment (adunsulag) [reply portal/home.php]: Should be fixed now.

File: portal/report/portal_patient_report.php (hunk 2 above)
Comment (adunsulag) [reply portal/report/portal_patient_report.php:48]: Should now be fixed.

File: src/Controllers/Portal/PortalPatientReportController.php (hunk 3 above)
Comment (adunsulag) [reply src/Controllers/Portal/PortalPatientReportController.php]: Removed, not used.

File: templates/portal/partial/reports/patient_report/_issues_encounters.html.twig (hunk 4 above)
Comment (adunsulag) [reply templates/portal/partial/reports/patient_report/_issues_encounters.html.twig]: Fixed.

File: templates/portal/partial/reports/patient_report/_issues_encounters_form.html.twig (hunk 5 above)
Comment (adunsulag) [reply templates/portal/partial/reports/patient_report/_issues_encounters_form.html.twig]: Fixed.

**Comment by adunsulag** [comment]
Fixed all the issues, if CI passes, will bring in to avoid holding up patch.

**Pull request merged**